src/pages/Explorer/block.state.ts

```diff
@@ -117,31 +118,56 @@ export const [blockInfo$, recordedBlocks$] = partitionByKey(
     ),
 )
 
-const getUnpinnedBlockInfo$ = (hash: string): Observable<BlockInfo> => {
-  const throughRpc$ = chainClient$.pipe(
+const getUnpinnedBlockInfo$ = (hash: string): Observable<BlockInfo> =>
+  client$.pipe(
     switchMap((client) =>
-      defer(() =>
-        client.client._request<
-          {
-            block: {
-              extrinsics: HexString[]
-              header: {
-                digest: { logs: Array<unknown> }
-                extrinsicsRoot: string
-                number: HexString
-                parentHash: HexString
-                stateRoot: HexString
-              }
-            }
-          } | null,
-          [string]
-        >("chain_getBlock", [hash]),
-      ).pipe(
-        repeat({
-          delay: 1000,
-        }),
+      from(client.getBlockHeader(hash)).pipe(
+        mergeMap((header) =>
+          combineLatest({
+            hash: of(hash),
+            parent: of(header.parentHash),
+            number: of(header.number),
+            body: client.watchBlockBody(hash),
+            events: from(
+              client.getUnsafeApi().query.System.Events.getValue({
+                at: hash,
+              }),
+            ),
+            header: from(client.getBlockHeader(hash)),
+            status: of(BlockState.Finalized),
+            diff: of(null),
+          }),
+        ),
+        catchError(() => getUnpinnedBlockInfoFallback$(hash, client)),
+        tap((v) => disconnectedBlocks$.next(v)),
       ),
```

Comment:
it's doing client.getBlockHeader() twice, one outside and another in the combineLatest - Maybe these can all be merged in a combine latest outside?

```suggestion
      combineLatest({
        header: client.getBlockHeader(hash),
        body: client.watchBlockBody(hash),
        events: client.getUnsafeApi().query.System.Events.getValue({
          at: hash,
        }),
      }).pipe(
        map(({ header, body, events }) => ({
          hash: hash,
          parent: header.parentHash,
          number: header.number,
          body,
          events,
          header,
          status: BlockState.Finalized,
          diff: null,
        })),
        catchError(() => getUnpinnedBlockInfoFallback$(hash, client)),
        tap((v) => disconnectedBlocks$.next(v)),
      ),
```

Reply:
UPS! thanks!